client/extensions/woocommerce/woocommerce-services/views/packages/index.js

```diff
@@ -1,61 +1,206 @@
+		} );
+	};
+
+	renderPredefHeader = ( isPlaceholder, title, selected, packages, serviceId, groupId ) => {
+		if ( isPlaceholder ) {
+			return (
+				<div className="packages__group-header" >
+					<BulkSelect
+						totalElements={ 0 }
+						selectedElements={ 1 }
+						className="packages__group-header-checkbox" />
+					<p /><p /><p />
+				</div>
+			);
+		}
+
+		if ( ! selected ) {
+			return null;
+		}
+
+		const onToggle = ( state, event ) => {
+			event.stopPropagation();
+			this.props.toggleAll( this.props.siteId, serviceId, groupId, event.target.checked );
+		};
+
+		return (
+			<div className="packages__group-header" >
+				<BulkSelect
+					totalElements={ packages.length }
+					selectedElements={ selected.length }
+					onToggle={ onToggle }
+					className="packages__group-header-checkbox" />
+				{ title }
+			</div>
+		);
+	};
+
+	renderPredefinedPackages = () => {
+		const elements = [];
+		const { siteId, isFetching, translate, form } = this.props;
+
+		if ( isFetching ) {
+			return [ {}, {} ].map( ( o, i ) => (
+				<FoldableCard
+					className="packages__predefined-packages placeholder"
+					key={ i }
+					header={ this.renderPredefHeader( true ) }
+					summary={ <p /> }
+					clickableHeader={ true }
+					expanded={ false }
+					icon="chevron-down"
+				/>
+			) );
+		}
+
+		forEach( form.predefinedSchema, ( servicePackages, serviceId ) => {
+			const serviceSelected = form.packages.predefined[ serviceId ] || [];
+
+			forEach( servicePackages, ( predefGroup, groupId ) => {
+				const groupPackages = predefGroup.definitions;
+				const nonFlatRates = reject( groupPackages, 'is_flat_rate' );
+				if ( ! nonFlatRates.length ) {
+					return;
+				}
+
+				const groupSelected = filter( serviceSelected, selectedId => some( groupPackages, pckg => pckg.id === selectedId ) );
+				const summary = this.predefSummary( groupSelected, nonFlatRates );
+
+				elements.push( <FoldableCard
+					className="packages__predefined-packages"
+					key={ `${ serviceId }_${ groupId }` }
+					header={ this.renderPredefHeader( false, predefGroup.title, groupSelected, nonFlatRates, serviceId, groupId ) }
+					summary={ summary }
+					expandedSummary={ summary }
+					clickableHeader={ true }
+					expanded={ false }
+					screenReaderText={ translate( 'Expand Services' ) }
+					icon="chevron-down"
+				>
+					<PackagesList
+						siteId={ siteId }
+						packages={ groupPackages }
+						selected={ groupSelected }
+						serviceId={ serviceId }
+						groupId={ groupId }
+						toggleAll={ this.props.toggleAll }
+						togglePackage={ this.props.togglePackage }
+						dimensionUnit={ form.dimensionUnit }
+						editable={ false } />
+				</FoldableCard> );
+			} );
+		} );
+
+		return elements;
+	};
+
+	render() {
+		const { isFetching, siteId, form, translate } = this.props;
+
+		const addPackage = () => ( this.props.addPackage( siteId ) );
+
 		return (
 			<div>
 				<ExtendedHeader
 					label={ translate( 'Packages' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **6** times:
`translate( 'Theme Packages' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).